packages/fcl/src/wallet-provider-spec/user-signature.md

```diff
@@ -0,0 +1,266 @@
+
+```javascript
+{
+	type: "FCL:FRAME:READY:RESPONSE",
+	body,
+	service: {
+	  params: service.params,
+	  data: service.data,
+	},
+}
+```
+```
+        FCL
+         |
+         |----------------------------+-[frame(service, body)]--+
+         |                            |                         |
+         |                            |         ENDPOINT        |
+         |                            |            |            |
+         |<------[message(FCL:FRAME:READY)]--------|            |
+         |                            |            |            |
+         |----------[message(FCL:FRAME:READY:RESPONSE)]-------->|  
+         |                            |            |            |
+         |<------[message(PollingResponse)]--------|            |
+         |                            |            |            |
+         |                            |                         |
+         |-----[CLOSE]--------------->+-------------------------+
+```
+ 
+*Additional query params can be added to the url based on details from user-signature service.*
+
+**Inside of this iframe, everything is up to the wallet**
+
+---
+
+Given this service:
+
+```javascript
+const service = {
+  f_type: "Service",
+  f_vsn: "1.0.0",
+  type: "user-signature",
+  uid: "uniqueDedupeKey",
+  endpoint: "http://rawr.io",
+  method: "IFRAME/RPC",
+  uid: "____",
+  endpoint: "https://___",
+  method: "IFRAME/RPC", // Service Methods: HTTP/POST | IFRAME/RPC
+  "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // wallets internal id for the user
+  data: {},   // included in body of user-signature request
+  params: {}, // included as query params on endpoint url
+}
+```
+
+When a user triggers `signUserMessage()`, and **FCL** has the above service.
+
+**FCL** will
+
+- Open an iframe at `http://rawr.io`
+- Post a message to the iframe
+- Wait for a response back that includes a status and relevant info
+- Handle response
+
+On receipt of **`FCL:FRAME:READY:RESPONSE`** the wallet should prompt user to approve, sign the message, and send back a response.
+
+## SIGNING THE MESSAGE
+
+If the user authorizes signing via wallet UI, wallet should hash and sign **`body.message`** before returning a PollingResponse with status **"APPROVED"** and a `CompositeSignature` as data.
+
+### UserDomainTag
+Before signing, the wallet may be required to add a specified DOMAIN TAG.  A domain tag is encoded as UTF-8 bytes, right padded to a total length of 32 bytes.
+
+The **`UserDomainTag`** is the prefix of all signed user space payloads.
+
+
+```javascript
+const rightPaddedHexBuffer = (value, pad) =>
+    Buffer.from(value.padEnd(pad * 2, 0), "hex")
+
+const USER_DOMAIN_TAG = rightPaddedHexBuffer(
+    Buffer.from("FLOW-V0.0-user").toString("hex"),
+    32
+  ).toString("hex")
+
+const prependUserDomainTag = msg => USER_DOMAIN_TAG + msg
+
+```
+
+**Example signing function**
+
+```javascript
```

Comment:
best not to get in the weeds here.
Would explain the domain tag.
Say it needs to be hashed using the strategy that is registered for the key on chain.
And that the signature in the composite signature needs to be a hex string.